sirepo/simulation_db.py

```diff
@@ -1151,11 +1264,18 @@ def _validate_name_uniquify(data, starts_with):
     data.models.simulation.name = n2
 
 
-
-# Ensure the value of a numeric field falls within the supplied limits (if any)
-# Note that currently the values in enum arrays at the indices below are sometimes
-# used for other purposes, so we return for non-numeric values rather than fail
 def _validate_number(val, sch_field_info):
+    """Ensure the value of a numeric field falls within the supplied limits (if any)
+
+    Note that currently the values in enum arrays at the indices below are sometimes
+    used for other purposes, so we return rather than fail for non-numeric values
+
+    Args:
+        val: numeric value to validate
+        sch_field_info ([str]): field info array from schema
+    Raises:
+        AssertionError if val is out of schema-specified range
+    """
     if len(sch_field_info) <= 4:
```

Comment:
Magic numbers are a problem here What is "4"?

Reply:
This is due to the info being an array; min goes in index 3, etc.  That will be dealt with by #1286 (I built the tool already but have not taken the plunge on converting everything)

Reply:
If the magic number were _FIELD_INFO['min'] = 4 and _FIELD_INFO['max'] = 5, then you would be able to refer to those inside _assert_float/field_info (see other comment) programmatically. You could then simplify the function signature further.